uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/GeneralSettings.php

```diff
@@ -0,0 +1,152 @@
+ */
+class CRM_HRLeaveAndAbsences_Form_GeneralSettings extends CRM_Core_Form {
+
+  /**
+   * @var array
+   *   The filter to pass to the setting.getfields API
+   */
+  private $_settingFilter = ['group' => 'leaveandabsence'];
+
+  /**
+   * @var array
+   *   An array to store settings once it has been retrieved from the settings API
+   */
+  private $_settings = [];
+
+  private $_submittedValues = [];
+
+  function buildQuickForm() {
+    $settings = $this->getFormSettings();
+
+    foreach ($settings as $name => $setting) {
+      if ($name == 'relationship_allowed_to_approve_leave') {
+        $this->$setting['html_type'](
+          $name,
+          [
+            'options' => $this->getRelationshipTypes(),
+            'multiple' => true,
+            'label' => $setting['label'],
+            'style' => $setting['html_attributes']['style'],
+          ],
+          true
+        );
+      }
+    }
+
+    $this->addButtons($this->getAvailableButtons());
+
+    // export form elements to template
+    $this->assign('elementNames', $this->getRenderableElementNames());
+
+    CRM_Core_Resources::singleton()->addStyleFile('uk.co.compucorp.civicrm.hrleaveandabsences', 'css/hrleaveandabsences.css');
+    parent::buildQuickForm();
+  }
+
+  function postProcess() {
+    $this->_submittedValues = $this->exportValues();
+    $this->saveSettings();
+
+    $url = CRM_Utils_System::url('civicrm/admin/leaveandabsences/generalsettings');
+    $session = CRM_Core_Session::singleton();
+    $session->replaceUserContext($url);
+  }
+
+  /**
+   * Get the fields/elements defined in this form.
+   *
+   * @return array
+   */
+  function getRenderableElementNames() {
+    // The _elements list includes some items which should not be
+    // auto-rendered in the loop -- such as "qfKey" and "buttons". These
+    // items don't have labels. We'll identify renderable by filtering on
+    // the 'label'.
+    $elementNames = [];
+    foreach ($this->_elements as $element) {
+      $label = $element->getLabel();
+      if (!empty($label)) {
+        $elementNames[] = $element->getName();
+      }
+    }
+    return $elementNames;
+  }
+
+  /**
+   * Get the settings we are going to allow to be set on this form.
+   *
+   * @return array
+   */
+  function getFormSettings() {
+    if (empty($this->_settings)) {
+      $settings = civicrm_api3('setting', 'getfields', ['filters' => $this->_settingFilter]);
+    }
+
+    return $settings['values'];
+  }
+
+  /**
+   * Save settings.
+   *
+   * @return array
+   */
+  function saveSettings() {
+    $values = array_intersect_key($this->_submittedValues, $this->getFormSettings());
+    civicrm_api3('setting', 'create', $values);
+  }
+
+  /**
+   * Set defaults for form.
+   *
+   * @see CRM_Core_Form::setDefaultValues()
+   */
+  function setDefaultValues() {
+    $existing = civicrm_api3('setting', 'get', ['return' => array_keys($this->getFormSettings())]);
```

Comment:
I notice you're always using 'setting' instead of 'Setting'. Can you please use the capitalized one in all calls?